UOP1_Project/Assets/Scripts/Characters/WalkingParticle.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class WalkingParticle : MonoBehaviour
+{
+
+    [Tooltip("Particle prefab to be attached to the charactor in the scene.")] public ParticleSystem walkingParticlePrefab;
+    [Tooltip("Minimum walking distance to play the particle.")] public float walkingDistance;
+
+    private Vector2 lastPosition;
+    private Vector2 currentPosition;
+    private ParticleSystem particleObject;
+
+    private void OnEnable()
+    {
+        lastPosition = new Vector2(transform.position.x, transform.position.z);
+        currentPosition = new Vector2(transform.position.x, transform.position.z);
+        if (!particleObject && walkingParticlePrefab)
```

Comment:
Alright, getting better :) What I would suggest now, is: there's no point having this behaviour be enabled if there's no particle prefab set up, right? It won't be able to do anything, it will just sit there wasting CPU cycles.

So, instead of saying "if there is a walkingParticlePrefab, then instantiate it" - the code will all be simpler if you say "if there is NOT a walkingParticlePrefab, then log an error and set `enabled = false;`. Setting enabled to false means that Update() will not run.

Once you have that in place, you can simplify Update() and remove the `if(particleObject)` check - you will be able to assume that there is always a particleObject, because you will have made it impossible to be running Update() without OnEnable() setting it up for you.